### PR TITLE
Task #8310: Handle event for Add Transaction Screen

### PR DESCRIPTION
**Pull request by tanbtd-1747**
### Issue Link :link:
https://edu-redmine.sun-asterisk.vn/issues/8310

### Goals :soccer:
Create a UI action for Add Transaction Screen
- Click Transaction Type Picker to go to Transaction Type Screen
- Click Transaction Date Picker to pop up DateTimePicker
- Click Save Button to go back to Dashboard Screen

Create a UI action for Transaction Type Screen
- Click corresponding segment on the Segmented Control to select income or expense types
- Click a row to go back to Add Wallet Screen

### Implementation Details :construction:
Change a little bit of the design. The timestamp label is replaced with the date picker

### CODE REVIEW

**Comment by tanbtd-1747** [comment]
ready

File: Money*/Money*/Scenes/TransactionType/TransactionTypeViewController.swift [hunk 1]
```diff
@@ -13,31 +13,72 @@ final class TransactionTypeViewController: UIViewController {
...
+                incomeTypes.append(transactionType.rawValue)
+            }
+        }
+        types = expenseTypes
+        transactionTypeTableView.reloadData()
+    }
+    
+    // MARK: - IBActions
+    @IBAction private func handleTransactionTypeSegmentedControlValueChanged(_ sender: UISegmentedControl) {
+        if sender.selectedSegmentIndex == 0 {
```
Comment (longvt96) [inline Money*/Money*/Scenes/TransactionType/TransactionTypeViewController.swift]:
```suggestion
      types = sender.selectedSegmentIndex == 0 ? expenseTypes : incomeTypes
```

File: Money*/Money*/Scenes/AddTransaction/AddTransactionViewController.swift [hunk 2]
```diff
@@ -16,23 +17,88 @@ final class AddTransactionViewController: UIViewController {
...
+        }
+        
+        guard let amount = amountTextField.text,
+            AccountValidator.validateIsNumber(amount) else {
+                presentErrorAlert(title: Constant.titleError,
+                                  message: Constant.messageTransactionErrorNotNumberBalance)
+                return
+        }
+        
+        transaction.name = name
```
Comment (tgdong2296) [inline Money*/Money*/Scenes/AddTransaction/AddTransactionViewController.swift]: cái này dùng `.do {` nếu có thể để cho clear code anh nhé

File: Money*/Money*/Scenes/TransactionType/TransactionTypeViewController.swift [hunk 3]
```diff
@@ -13,31 +13,72 @@ final class TransactionTypeViewController: UIViewController {
...
+            types = expenseTypes
```
Comment (tgdong2296) [inline Money*/Money*/Scenes/TransactionType/TransactionTypeViewController.swift]:
```suggestion
            types = sender.selectedSegmentIndex == 0 ? expenseTypes : incomeTypes
```

**Comment by tanbtd-1747** [comment]
ready

**Comment by tanbtd-1747** [comment]
ready

**Review by longvt96 — APPROVED** [review]

**Review by tgdong2296 — APPROVED** [review]

**Review by tienpx3010 — APPROVED** [review]

**Pull request merged**